### PR TITLE
Update CI config

### PR DESCRIPTION
**Pull request by Cadair**
This is some changes made in https://github.com/sunpy/ndcube/pull/747 but template side.

Changes to the GitHub actions workflows need manual re-templating, so if there are any other tweaks we want to make to this file let's batch them together.

### CODE REVIEW

File: {{ cookiecutter.package_name }}/.github/workflows/ci.yml [hunk 1]
```diff
@@ -90,21 +90,17 @@ jobs:
...
 
-{% if cookiecutter.use_compiled_extensions == 'y' -%}
   publish:
-    # Build wheels when pushing to any branch except main
-    # publish.yml will only publish if tagged ^v.*
+    needs: [test, docs]
+    # Build wheels on PRs only when labelled. Releases will only be published if tagged ^v.*
+    # see https://github-actions-workflows.openastronomy.org/en/latest/publish.html#upload-to-pypi
     if: |
+      github.event_name != 'pull_request' ||
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/.github/workflows/ci.yml:98]: I think it's probably overkill to skip wheel builds on main? For projects which do releases off main it's actively harmful, and for projects which use branches it's not a big deal?

File: {{ cookiecutter.package_name }}/.github/workflows/ci.yml [hunk 2]
```diff
@@ -90,21 +90,17 @@ jobs:
...
-        github.event_name != 'pull_request' && (
-          github.ref_name != 'main' ||
-          github.event_name == 'workflow_dispatch'
-        )
-      ) || (
         github.event_name == 'pull_request' &&
         contains(github.event.pull_request.labels.*.name, 'Run publish')
       )
-    needs: [test, docs]
+{% if cookiecutter.use_compiled_extensions == 'y' -%}
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/.github/workflows/ci.yml]: @CyclingNinja We need to double check that moving some of this out of the if statement hasn't broken the whitespace.

File: {{ cookiecutter.package_name }}/.github/workflows/ci.yml (hunk 1 above)
Comment (nabobalis) [reply {{ cookiecutter.package_name }}/.github/workflows/ci.yml:98]: I have no issue with building wheels on branches or main. I think its better than finding out during a release.

**Comment by nabobalis** [comment]
You mean other CI changes or in general?

**Review by nabobalis — APPROVED** [review]

**Comment by Cadair** [comment]
I just need to verify the whitespace changes before we merge this.

**Comment by Cadair** [comment]
> You mean other CI changes or in general?

specifically CI as they can't be rolled out automatically.

File: {{ cookiecutter.package_name }}/.github/workflows/ci.yml (hunk 2 above)
Comment (Cadair) [reply {{ cookiecutter.package_name }}/.github/workflows/ci.yml]: It had, but I fixed it with the new test stuff.

File: tests/conftest.py [hunk 3]
```diff
@@ -88,5 +88,13 @@ def bake_examples_compiled_dev_version(cookies):
...
                                          "use_compiled_extensions": "y",
                                          "enable_dynamic_dev_versions": "y",
-                                         "author_name": "test"})
+                                         "author_name": "test",
+                                         "include_cruft_update_github_workflow": "y",
+                                         })
     return _handle_cookiecutter_errors(result)
+
+
+# Add support for saving out rendered cookies to a specific locatiuuon for inspection
```
Comment (Cadair) [inline tests/conftest.py]:
```suggestion
# Add support for saving out rendered cookies to a specific location for inspection
```

File: {{ cookiecutter.package_name }}/pyproject.toml [hunk 4]
```diff
@@ -6,8 +6,8 @@ requires = [
 {%- if cookiecutter.use_compiled_extensions == 'y' %}
   "extension-helpers",
   "numpy>=1.25",
-  "cython"
-{% endif -%}
+  "cython",
```
Comment (CyclingNinja) [inline {{ cookiecutter.package_name }}/pyproject.toml:9]: should the comma be in the % magic? are we okay with floating `,`?

Comment (Cadair) [reply {{ cookiecutter.package_name }}/pyproject.toml:9]: Trailing comma was the intention.

**Review by CyclingNinja — APPROVED** [review]

File: README.rst [hunk 5]
```diff
@@ -1,5 +1,10 @@
...
 
-The `OpenAstronomy Python Packaging Guide <https://packaging-guide.openastronomy.org/en/latest/>`__ that contains an embedded `cookiecutter <https://cookiecutter.readthedocs.io/>`__ template.
-Please see also the `instructions for using the template <https://packaging-guide.openastronomy.org/en/latest/#using-the-template>`__.
+This repo extends the `OpenAstronomy Python Packaging Guide <https://packaging-guide.openastronomy.org/en/latest/>`__ to add SunPy specific features such as config files which are shared over all of SunPy's packages.
+
+Previewing Rendered Templates
+-----------------------------
+
+To make it easier to preview changes to files when rendering the template there is a tox environment named ``bake_cookies``.
+If you run ``tox -e bake_cookies`` a new directory named ``cookies`` will be present with a variety of test renders of the template in (see ``tests/conftest.py`` for the configuration options).
```
Comment (nabobalis) [inline README.rst:10]: Does the ``--cookie-location`` work if that is passed to tox?

I ask as I have a similar problem with magex and I assume its a config issue.

File: tests/conftest.py [hunk 6]
```diff
@@ -88,5 +88,13 @@ def bake_examples_compiled_dev_version(cookies):
...
+                                         "author_name": "test",
+                                         "include_cruft_update_github_workflow": "y",
+                                         })
     return _handle_cookiecutter_errors(result)
+
+
+# Add support for saving out rendered cookies to a specific location for inspection
+
+def pytest_addoption(parser):
+    parser.addoption("--cookie-location", action="store", default=None)
```
Comment (nabobalis) [inline tests/conftest.py]:
```suggestion
def pytest_addoption(parser):
    # Add support for saving out rendered cookies to a specific location for inspection
    parser.addoption("--cookie-location", action="store", default=None)
```

File: README.rst (hunk 5 above)
Comment (Cadair) [reply README.rst:10]: It is just a pytest CLI arg so it passes through like normal.

**Comment by nabobalis** [comment]
Having a look over the other CI files. Nothing else screams out to me.

File: README.rst (hunk 5 above)
Comment (nabobalis) [reply README.rst:10]: Sure, but does it get parsed correctly. In the magex case, pytest says its not valid.

**Pull request merged**